mender.go

```diff
@@ -56,49 +58,72 @@ type MenderState int
 //       v                     v
 // fresh update         wait for update
 //
-// Any state can transition to MenderStateError, setting LastError()
-// to the error that triggered the transition
+// Any state can transition to MenderStateError
 
 const (
-	MenderStateUnknown MenderState = iota
 	// initial state
-	MenderStateInit
+	MenderStateInit MenderState = iota
 	// client is bootstrapped, i.e. ready to go
 	MenderStateBootstrapped
-	// update applied, waiting for commit
-	MenderStateRunningWithFreshUpdate
 	// wait for new update
-	MenderStateWaitForUpdate
-	// error occurred, call Controller.LastError() to obtain the
+	MenderStateUpdateCheckWait
+	// check update
+	MenderStateUpdateCheck
+	// update fetch
+	MenderStateUpdateFetch
+	// update install
+	MenderStateUpdateInstall
+	// commit needed
+	MenderStateUpdateCommit
+	// reboot
+	MenderStateReboot
 	// error
 	MenderStateError
+	// exit state
+	MenderStateDone
 )
 
 type mender struct {
-	BootEnvReadWriter
-	state          MenderState
-	config         menderFileConfig
+	UInstallCommitRebooter
```

Comment:
Anonymous `MenderPieces` can be a part of mender instead of repeating all those pieces again here.